internal/slicer/slicer.go

```diff
@@ -325,44 +284,105 @@ func Run(options *RunOptions) (*Report, error) {
 		}
 	}
 
+	err := removeUntilMutate(targetDirAbs, knownPaths)
+	if err != nil {
+		return nil, err
+	}
+
+	return report, nil
+}
+
+// removeUntilMutate removes entries marked with until: mutate. A path is marked
+// only when all slices that refer to the path mark it with until: mutate.
+func removeUntilMutate(rootDir string, knownPaths map[string]pathData) error {
```

Comment:
s/removeUntilMutate/removeAfterMutate/

At the moment it's reading reversed: it's actually removing now that files have already been mutated, instead of before (and until) they are mutated.

Reply:
Done in https://github.com/canonical/chisel/pull/136/commits/7c4efa2f662a5ce54595f844bf43a35288509318.